stable/database/files/nuosm

```diff
@@ -387,6 +387,69 @@ function resolveLatestSource() {
   return 0
 }
 
+#=======================================
+# function - Looks for a previous archive and journal and imports it
+#
+function loadFromSnapshot() {
+  local RECREATE_ARCHIVES="false"
+
+  if [ -z "$(ls -A ${DB_DIR})" ]; then
+    local archives=$( find "/var/opt/nuodb/archive" -name info.json )
+    if [ -z "$archives" ] || [ $(echo "$archives" | wc -l) != 1 ]; then 
+      die 1 "Did not find exactly 1 archive: ${archives}"
+    fi
+
+    local ARCHIVE_PATH=$( dirname $archives )
+    log "Found snapshot at ${ARCHIVE_PATH}"
+
+    if [ -n "$BACKUP_ID" ]; then
+      if [ ! -e ${ARCHIVE_PATH}/backup.json ] || \
+          [ $BACKUP_ID != $(sed -e 's|.*"id":\s*"\?\([^\s"]\+\)"\?[,}].*|\1|' "$ARCHIVE_PATH/backup.json") ]; then
```

Comment:
If the `backup.json` file has an extra space before the colon as below, then this will not find the backup ID.
```json
{
  "id" : "theId"
}
```

I think it defeats the point of using a JSON format if it is brittle like this. My preference is to just use a file with its entire contents consisting of the backup ID, and we can migrate to a JSON format whenever we `nuosm` as a `nuocmd` subcommand in Python.

Reply:
I really like the idea of having it be a JSON file, for future extension. I was a bit in denial about how brittle the regex would be.

You are right that it is probably too much complexity for no immediate benefit.